src/domain/Lists/DraggableList/DraggableList.tsx

```diff
@@ -0,0 +1,84 @@
+import React from 'react'
+import {
+  DragDropContext,
+  Draggable,
+  DropResult,
+  Droppable
+} from 'react-beautiful-dnd'
+import uuid from 'uuid'
+
+import { Props } from '../../../common'
+import { StyledListArea, StyledListItem } from './style'
+
+interface IDraggableListProps<T> {
+  /** The array of items */
+  items: T[]
+  /** Function to update the array of items when they have been re-arranged */
+  setItems: (nodes: T[]) => void
+  /** An array of ReactNode that should represent the array of items */
+  children: React.ReactNode[]
+  /** The data-component-context */
+  componentContext?: string
+}
+
+const DraggableList = <T extends {}> ({ items, setItems, children, componentContext }: IDraggableListProps<T>) => {
+  const onDragEnd = (result: DropResult) => {
+    if (!result.destination) {
+      return
+    }
+    const { source, destination } = result
+
+    const newItems = [...items]
+    const [removed] = newItems.splice(source.index, 1)
+    newItems.splice(destination.index, 0, removed)
+    setItems(newItems)
```

Comment:
So to do this without using `items` it will look something like this:
```
setItems(previousItems => 
    const newItems = previousItems
    const [removed] = newItems.splice(source.index, 1)
    newItems.splice(destination.index, 0, removed)
    return newItems
)
```